packages/contracts/src/utils/JwtVerifier.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "../interfaces/IJwtGroth16Verifier.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { strings } from "solidity-stringutils/src/strings.sol";
+
+struct EmailProof {
+    string domainName; // Domain name of the sender's email
+    bytes32 publicKeyHash; // Hash of the DKIM public key used in email/proof
+    uint timestamp; // Timestamp of the email
+    string maskedCommand; // Masked command of the email
+    bytes32 emailNullifier; // Nullifier of the email to prevent its reuse.
+    bytes32 accountSalt; // Create2 salt of the account
+    bool isCodeExist; // Check if the account code is exist
+    bytes proof; // ZK Proof of Email
+}
+
+contract JwtVerifier is OwnableUpgradeable, UUPSUpgradeable {
+    using strings for *;
+
+    IJwtGroth16Verifier groth16Verifier;
+
+    uint256 public constant ISS_FIELDS = 2;
+    uint256 public constant ISS_BYTES = 32;
+    uint256 public constant COMMAND_FIELDS = 20;
+    uint256 public constant COMMAND_BYTES = 605;
+    uint256 public constant AZP_FIELDS = 1;
+    uint256 public constant AZP_BYTES = 14;
+
+    constructor() {}
+
+    /// @notice Initialize the contract with the initial owner and deploy Groth16Verifier
+    /// @param _initialOwner The address of the initial owner
+    function initialize(
+        address _initialOwner,
+        address _groth16Verifier
+    ) public initializer {
+        __Ownable_init(_initialOwner);
+        groth16Verifier = IJwtGroth16Verifier(_groth16Verifier);
+    }
+
+    function verifyEmailProof(
+        EmailProof memory proof
+    ) public view returns (bool) {
+        (
+            uint256[2] memory pA,
+            uint256[2][2] memory pB,
+            uint256[2] memory pC
+        ) = abi.decode(proof.proof, (uint256[2], uint256[2][2], uint256[2]));
+
+        uint256[ISS_FIELDS + COMMAND_FIELDS + AZP_FIELDS + 6] memory pubSignals;
+        
+        // string[] = [kid, iss, azp]
+        string[] memory parts = this.stringToArray(proof.domainName);
```

Comment:
Can you please explain how this encodes the inputs?

Reply:
@shreyas-londhe I added the comments